### PR TITLE
Add generated list of all actions

### PR DESCRIPTION
**Pull request by arielshaqed**
* Auto-generate list (slice) of actions by examining all consts in package permissions.
* Add the generating tool, required changes to Makefile.
* Test that the list is reasonable.

Needed as part of #5371.

### CODE REVIEW

File: cmd/tools/extract_actions/main.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	"go/types"
+	"log"
+	"os"
+)
+
+func main() {
+	inFile := os.Args[1]
+	outFile := os.Args[2]
+
+	out, err := os.Create(outFile)
```
Comment (nopcoder) [inline cmd/tools/extract_actions/main.go:18]: Move the creation after the inFile processing

File: cmd/tools/extract_actions/main.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package main
```
Comment (nopcoder) [inline cmd/tools/extract_actions/main.go:1]: Suggest adding a build constraint + move the file to the same package (as, for example: codegen.go)
```go
//go:build gen
```

File: pkg/permissions/actions.go [hunk 3]
```diff
@@ -1,3 +1,4 @@
+//go:generate go run github.com/treeverse/lakefs/cmd/tools/extract_actions ./actions.go ./actions.gen.go
```
Comment (nopcoder) [inline pkg/permissions/actions.go:1]: Based on previous suggestion we can just run the code 
```go
//go:generate go run codegen.go -i actions.go -o actions.gen.go
```

**Review by nopcoder — APPROVED** [review]
LGTM :) couple of suggestions, not mandatory

File: cmd/tools/extract_actions/main.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	"go/parser"
+	"go/token"
+	"go/types"
+	"log"
+	"os"
+)
+
+func main() {
+	inFile := os.Args[1]
+	outFile := os.Args[2]
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply cmd/tools/extract_actions/main.go:16]: Not done, not an important enough tool.

File: cmd/tools/extract_actions/main.go (hunk 1 above)
Comment (arielshaqed) [reply cmd/tools/extract_actions/main.go:18]: Not done: if I cannot create the file, that's the first thing I should complain about!

File: cmd/tools/extract_actions/main.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+	out, err := os.Create(outFile)
+	if err != nil {
+		log.Fatal(err)
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply cmd/tools/extract_actions/main.go:20]: Done!

File: cmd/tools/extract_actions/main.go [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+	constantValues := []string{}
+	ConstantsOf(inFile, func(v string) {
+		constantValues = append(constantValues, v)
+	})
+
+	fmt.Fprintln(out, `// Code generated by extract_actions. DO NOT EDIT.
+package permissions
+var Actions = []string{`)
+	for _, c := range constantValues {
+		fmt.Fprintln(out, c, ",")
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply cmd/tools/extract_actions/main.go:32]: Done!

**Review by arielshaqed — COMMENTED** [review]
Thanks!

Moving the file under a separate response, to make it easier to follow what happened in each commit!

**Review by arielshaqed — COMMENTED** [review]
Moved!

Pulling once tests pass, thanks!

**Pull request merged**